clients/src/main/java/org/apache/kafka/common/protocol/Errors.java

```diff
@@ -479,6 +482,19 @@ public static Errors forException(Throwable t) {
         return UNKNOWN_SERVER_ERROR;
     }
 
+    public static Throwable getCause(Throwable t) {
```

Comment:
This name might be a little confusing since `Throwable#getCause` exists.

Reply:
Yeah, let me try to think of a better name. One idea I was toying with is letting this method search the exception chain for the first ApiException. Then it doesn't need to be specific to CompletionException or ExecutionException. What do you think about that?

Reply:
I decided not to do this. There may be cases where we have intentionally wrapped an exception type because we don't want to expose the underlying error to the client.